hazelcast/src/main/java/com/hazelcast/config/ConfigXmlGenerator.java

```diff
@@ -1070,8 +1067,10 @@ private static void cacheConfigXmlGenerator(XmlGenerator gen, Config config) {
 
             gen.node("merge-policy", c.getMergePolicyConfig().getPolicy());
             appendEventJournalConfig(gen, c.getEventJournalConfig());
-            appendHotRestartConfig(gen, c.getHotRestartConfig());
-            appendMerkleTreeConfig(gen, c.getMerkleTreeConfig());
+            appendDataPersistenceConfig(gen, c.getDataPersistenceConfig());
+            if (c.getMerkleTreeConfig().getEnabled() != null) {
+                appendMerkleTreeConfig(gen, c.getMerkleTreeConfig());
+            }
```

Comment:
I think, this will break the equality of generated configs. (Config1->XML->Config2)

Reply:
Pushed tests to verify equality is not broken. If user did not touch `MerkleTreeConfig`, no `merkle-tree` element is generated in XML output -> when read from XML `enabled` stays `null` which is same as before transformation to XML.